samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -110,6 +110,10 @@ public StreamProcessor(Config config, Map<String, MetricsReporter> customMetrics
     this(config, customMetricsReporters, (Object) streamTaskFactory, processorListener, null);
   }
 
+  public String toString() {
```

Comment:
It is useful for debugging. I am not sure where else can it be defined.
We cannot have it package private because it overrides a public method.